packages/web-app-files/src/components/SideBar/Shares/InviteCollaborator/ExpirationDatepicker.vue

```diff
@@ -50,105 +50,112 @@ import { DateTime } from 'luxon'
 export default {
   name: 'ExpirationDatepicker',
   props: {
-    expirationDate: {
-      type: Date,
+    shareTypes: {
+      type: Array,
       required: false,
-      default: undefined
+      default: () => []
     }
   },
 
   data() {
     return {
-      enteredExpirationDate: null
+      expirationDateInternal: null
     }
   },
 
   computed: {
     ...mapGetters(['capabilities']),
 
-    expirationSupported() {
-      return this.userExpirationDate && this.groupExpirationDate
-    },
-
-    defaultExpirationDateSet() {
-      return this.userExpirationDate.enabled || this.groupExpirationDate.enabled
+    expirationDate: {
+      get: function () {
+        return this.expirationDateInternal || this.expirationDateDefault
+      },
+      set: function (val) {
+        this.expirationDateInternal = val
+
+        const expirationDate = DateTime.fromJSDate(val)
+          .setLocale(this.$language.current)
+          .endOf('day')
+        this.$emit('optionChange', {
+          expirationDate: expirationDate.isValid
+            ? expirationDate.toFormat("yyyy-MM-dd'T'HH:mm:ssZZZ")
+            : null
+        })
+      }
     },
 
-    userExpirationDate() {
-      return this.capabilities.files_sharing.user.expire_date
-    },
+    expirationDateDefault() {
+      if (!(this.expirationDateUser.enabled || this.expirationDateGroup.enabled)) {
+        return null
+      }
 
-    groupExpirationDate() {
-      return this.capabilities.files_sharing.group.expire_date
-    },
+      const hasUserCollaborators = this.shareTypes.includes(0)
+      const hasGroupCollaborators = this.shareTypes.includes(1)
+      const userMaxExpirationDays = parseInt(this.expirationDateUser.days, 10)
+      const groupMaxExpirationDays = parseInt(this.expirationDateGroup.days, 10)
 
-    defaultExpirationDate() {
-      if (!this.defaultExpirationDateSet) {
-        return null
+      let days = null
```

Comment:
i adopted now the style from the prev version, but i had to extend it because we now also respect if the component has a user or/and a group.

if group has a max of 3 and user a max of 2:
- user selected 14 days from now, then adds a group it changes to 3 days from now
- user then also adds a user, then it changes dynamic to 2